Add examples for using dimensionless units

tcjansen opened the pull request:
As a user, I had some confusion when trying to use and create dimensionless quantities, and according to issue #5850 I'm not the only one. To help others like me, I've added an example of creating a dimensionless quantity and a link to the more detailed dimensionless quantities section to the top-level "Getting Started" astropy.units documentation. 

In the more detailed section on dimensionless quantities, I've also added an example of creating a _scaled_ dimensionless quantity and why you might want to do so (see the second half of [this comment](https://github.com/astropy/astropy/issues/5850#issuecomment-283091582)).

Let me know if my changes can be improved in any way!

EDIT: Fix #5850 

cc @bmorris3 and @eteq

pllim commented on docs/units/standard_units.rst:
```diff
@@ -162,7 +162,26 @@ For example::
-
+   
+To create a dimensionless quantity with a specific scaling factor, 
+multiply or divide a value by the ``Unit()`` method with the scaling 
```
Technically, `Unit()` is not a method:
```suggestion
multiply or divide a value by ``Unit()`` with the scaling 
```

tcjansen replied:
Ah okay thanks!

bsipocz commented on docs/units/index.rst:
```diff
@@ -82,6 +82,15 @@ Units that "cancel out" become a special unit called the
+or divide a value by the unscaled dimensionless unit::
+
+    >>> q1 = 1.0 * u.dimensionless_unscaled
+    >>> # or equivalently, though less preferred for readability:
```
leave comments out of code example, in general examples should show one new behaviour, stick with one of them or add two separate examples.

tcjansen replied:
Good to know, thanks

tcjansen commented:
> For future reference, please don't do `git pull`, as that creates an unnecessary "merge commit" in your git log that would make backporting difficult for the release manager. 

Okay, thanks for letting me know @pllim

mhvk commented on docs/units/index.rst:
```diff
@@ -82,6 +82,13 @@ Units that "cancel out" become a special unit called the
     Unit(dimensionless)
+    
+To create a simple :ref:`dimensionless quantity <doc_dimensionless_unit>`, multiply 
+or divide a value by the unscaled dimensionless unit::
```
I'd delete "or divide" here - it is true strictly speaking but more likely to confuse than to help (really, nobody should do `1 / u.dimensionless_unscaled`!

tcjansen replied:
Implemented!

mhvk commented on docs/units/standard_units.rst:
```diff
@@ -162,7 +162,26 @@ For example::
-
+   
+To create a dimensionless quantity with a specific scaling factor, 
+multiply or divide a value by ``Unit()`` with the scaling 
```
Here again I would just describe multiplication, and write `Unit(scale_factor)`. I'd also give the thing we do *not* recommend in the text rather than as an actual example, maybe as `mega_factor = 20`.

mhvk replied:
Also, might it make sense to tell that `u.percent` is implemented exactly like this?

mhvk commented on docs/units/standard_units.rst:
```diff
@@ -162,7 +162,26 @@ For example::
+
+   >>> big_unitless_num = 20000000 # 20 million
+
+   >>> # I will be doing many calculations with big_unitless_num, 
```
Generally, I would not put further comments in the example proper; just move it to the main text.

mhvk commented on docs/units/standard_units.rst:
```diff
@@ -162,7 +162,26 @@ For example::
+   >>> # The above will work, but I want to keep track of this scaling
+   >>> # factor through all of my calculations. u.Unit() can do this
+   >>> # for us:
+   >>> big_unitless_num *= u.Unit(1e6)
```
I think this is clearer as a new definition of the number, i.e.,
```
big_unitless_num = 20 * u.Unit(1e6)
```
Partly, this is because the in-place operation would not work well if the number was an array instead (long-standing bug, difficult to fix...).

bsipocz commented:
This still needs a rebase. As @pllim said above that first merge commit of a PR that updates the fork's master should be removed in favour of following the development workflow.

tcjansen commented:
> @tcjansen - thanks for the work so far. Some further comments inline.

Thanks for the suggestions @mhvk - I moved all of the comments out of the example code and into the text. I'm a little worried that it's a bit verbose but I'm not sure how to get the same point across in less words. But if it looks okay to you all then it's okay with me!

tcjansen commented:
> This still needs a rebase. As @pllim said above that first merge commit of a PR that updates the fork's master should be removed in favour of following the development workflow.

Sorry - still getting acquainted with github and contributing! I believe I've rebased and squashed correctly now...

mhvk commented on docs/units/index.rst:
```diff
@@ -82,6 +82,12 @@ Units that "cancel out" become a special unit called the
     >>> u.m / u.m
     Unit(dimensionless)
+    
+To create a simple :ref:`dimensionless quantity <doc_dimensionless_unit>`, multiply a value by the unscaled dimensionless unit::
```
Nitpick, but could you break this line so that it is no more than 80 characters? Otherwise, this looks all OK.

tcjansen replied:
Oops, slipped by me.

mhvk commented on docs/units/standard_units.rst:
```diff
@@ -162,7 +162,24 @@ For example::
+
+   >>> some_measurement = 5.0 * u.cm
+   >>> some_measurement * big_unitless_num # answer is 100 million cm
+   <Quantity 100. 1e+06 cm>
```
The travis failure for this is real, though. Could you add `# doctest: +FLOAT_CMP` to the line above? (probably OK after the comment you already have there).

bsipocz replied:
there is no need for the comment though. Also in either case please set up your editor so that it notifies you of not following PEP8. In some cases it's perfectly fine not to do so, but here, before inline comments there should be 2 spaces.

mhvk commented:
This looks all OK now, so merging! Thanks, @tcjansen for sticking with it!
